diff --git a/vold/fs/Vfat.cpp b/vold/fs/Vfat.cpp
index 9878a53..4c3c5f6 100644
--- a/vold/fs/Vfat.cpp
+++ b/vold/fs/Vfat.cpp
@@ -133,7 +133,7 @@
     const char* c_source = source.c_str();
     const char* c_target = target.c_str();
 
-    flags = MS_NODEV | MS_NOSUID | MS_DIRSYNC;
+    flags = MS_NODEV | MS_NOSUID ;
 
     flags |= (executable ? 0 : MS_NOEXEC);
     flags |= (ro ? MS_RDONLY : 0);

